docs/static/hardware.rst

```diff
@@ -0,0 +1,271 @@
+:code:`char`         :code:`uint8_t`, :code:`int8_t`        8          1          8/1           0 to 255                     -128 to 127
+:code:`short`        :code:`uint16_t`, :code:`int16_t`      16         2          16/2          0 to 65535                   -32768 to 32767
+:code:`int`          :code:`uint24_t`, :code:`int24_t`      24         3          32/4          0 to 16777215                -8388608 to 8388607
+:code:`long`         :code:`uint32_t`, :code:`int32_t`      32         4          32/4          0 to 4294967295              -2147483648 to 2147483647
+:code:`long log`     :code:`uint64_t`, :code:`int64_t`      64         8          64/8          0 to a lot                   -negative lot to a lot
+:code:`void*`        :code:`uintptr_t`, :code:`ptrdiff_t`   24         3          64/8          All of RAM and the archive   N/A
+==================   ====================================   =======    ========   ===========   ==========================   ==========================
+
+"Common Size" is the size in bit/bytes of the type that Windows uses, as well as many other platforms;
+on some platforms, the types are larger than in this toolchain.
+
+Floating-Point Numbers
+^^^^^^^^^^^^^^^^^^^^^^
+
+Perhaps surprisingly, even though the Z80 is used in a calculator, the CPU itself does not have any concept of :code:`float` or :code:`double`.
+The circuits that modern CPUs use to handle such types is large and complicated, and simply wasn't feasible in the 70s when the Z80 was designed.
+Therefore, software has to implement that functionality.
+
+Like the Z80, the eZ80 also lacks native support for float-point arithmetic, so :code:`float` is slower than :code:`int`.
+Furthermore, in the toolchain, :code:`float` and :code:`double` are actually treated the same (this is allowed by the C standard).
+
+In the toolchain, :code:`float` is the standard IEEE 32-bit float.
+However, IEEE floats have limited precision and have unintuitive rouding behavior.
+So when TI designed the TI-81, they wrote their own special floating-point number format, which is much more suitable for precision scientific calculations.
+Unfortunately, it also much slower.
+
+This, however, has nothing to do with the calculator's hardware, and is instead covered on a different page.
+
+The Memory Map
+^^^^^^^^^^^^^^
+
+Both flash and RAM are mapped into the eZ80's address space.
+This means that a pointer may point to RAM or flash (or something else!) depending on its value.
+
+====================================    ===============
+Pointer Value                           Memory Accessed
+====================================    ===============
+:code:`0x000000` to :code:`0x3FFFFF`    Flash
+:code:`0x400000` to :code:`0xCFFFFF`    Nothing
+:code:`0xD00000` to :code:`0xD3FFFF`    RAM
+:code:`0xD40000` to :code:`0xD657FF`    VRAM
+:code:`0xD65800` to :code:`0xFFFFFF`    I/O
```

Comment:
Similarly to my comment above, 0xD65800 to 0xDFFFFF has some stuff going on, but it might be beyond the scope of this document.